java/code/src/com/suse/manager/reactor/PGEventStream.java

```diff
@@ -132,40 +147,50 @@ public void run() {
 
     @Override
     public void notification(int processId, String channelName, String payload) {
-        notification(Long.parseLong(payload));
+        List<Long> counts = Arrays.stream(payload.split(","))
+                .map(str -> Long.valueOf(str))
+                .collect(Collectors.toList());
+        notification(counts);
     }
 
     /**
      * Called every time a notification from Postgres (ultimately from mgr_engine.py) is fired.
-     * @param count the number of INSERTed events to be handled
+     * @param counts the list of the number of INSERTed events to be handled by queue. The first queue
+     *      is the one for events that aren't associated to a minion, queue 0.
      */
-    public void notification(long count) {
+    public void notification(List<Long> counts) {
+        LOG.trace("Got notification: " + counts);
         // compute the number of jobs we need to do - each job COMMITs individually
         // jobs = events / MAX_EVENTS_PER_COMMIT (rounded up)
-        long jobs = (count + MAX_EVENTS_PER_COMMIT - 1) / MAX_EVENTS_PER_COMMIT;
-
-        // queue one handlingTransaction(processEvents) call per job
-        for (int j = 0; j < jobs; j++) {
-            executorService.execute(() -> {
-                List<SaltEvent> uncommittedEvents = new LinkedList<>();
-                TransactionHelper.handlingTransaction(
-                        () -> processEvents(uncommittedEvents),
-                        e -> handleExceptions(uncommittedEvents, e));
-            });
+        for (final int queue : IntStream.range(0, THREAD_POOL_SIZE + 1).toArray()) {
```

Comment:
Is there a particular reason not to use `IntStream.range(...).forEach()`?

Reply:
Just because I was hibernating when Java8 went out may be?

Reply:
:heavy_check_mark: